### DISCUSSION TITLE
Any examples of setting up csla for DI in Asp.net on .net framework?

### DISCUSSION THREAD

**Discussion by ajohnstone-ks** [MarimerLLC/csla#1824] [Questions] [answered]
In the example I see how Csla is configured to use DI in the data portal methods, but these are using Asp.net Core, and I'm still on .net framework.

Any examples of setting up Csla for DI on .net framework?  Ideally using Unity for the container 😄

**Comment by rockfordlhotka** [comment]
I am pretty sure the WfpUI project in the ProjectTracker solution currently demonstrates how to initialize DI during startup.

It is a little tricky, from what I've found, to set up DI in WinForms and WPF - at least for .NET Framework projects - but it can be done.

There are lots of examples of console app setup, and of course ASP.NET setup out there in the wild.

I haven't ever looked at how to swap out the default DI engine provided by .NET for something like Unity or Autofac. I have read articles that indicate these things are possible (at least with Autofac), because there's a type of plug-in model behind the Microsoft DI framework.

**Reply by rockfordlhotka** [reply]
Oh, I missed that you were looking at ASP.NET on Framework.

What I'd recommend is creating a new web project using ASP.NET and targeting .NET Framework 4.8. The Microsoft project template should show you how to do everything (other than Unity).

**Reply by ajohnstone-ks** [reply]
I'll have to check out the template, but I did get it working.

**Comment by ajohnstone-ks** [comment, marked as answer]
In case anyone else needs this, I think its as simple as:

```
// Configure your container above
var provider = new Unity.Microsoft.DependencyInjection.ServiceProviderFactory(container);
Csla.ApplicationContext.DefaultServiceProvider = provider.CreateServiceProvider(container);
```

I needed to add Microsoft.Extensions.DependencyInjection and Unity.Microsoft.DependencyInjection nuget packages.  Once  I did that, dependencies were being correctly injected into the data portal methods.
